dlk/python/dlk/plugins/tf.py

```diff
@@ -241,150 +223,74 @@ class Input(object):
         """Get shape info."""
-        if self.is_placeholder:
-            return [d.size for d in self.in_.attr.get('shape').shape.dim]
+        shape_sets: List = None
+        if len(self.attrs_to_value.keys() & {'_output_shapes', 'shape'}) > 0:
+            for x in self.attrs_to_value.keys() & {'_output_shapes', 'shape'}:
+                if self.attrs_to_value[x]:
+                    shape_sets = self.attrs_to_value[x]
+        if shape_sets is not None:
+            return shape_sets
+        elif self.attrs_to_value['value'].tensor_shape.dim:
+            return [d.size for d in self.attrs_to_value['value'].tensor_shape.dim]
         else:
-            return [d.size for d in self.tensor.tensor_shape.dim] or [self.get_data().size]
-
-    def set_shape(self, val: List[str]) -> None:
-        """Set shape info."""
-        raise NotImplemented
-
-
-class Output(object):
-    def __init__(self, out_nd) -> None:  # type: ignore
-        self.out_ = out_nd
-
-    @property
-    def name(self) -> str:
-        """Return the name corresponding to the node."""
-        return self.out_.name.replace('/', '_').replace('-', '_')
+            return [self.get_data().size]
 
-    @property
-    def op_type(self) -> str:
-        """Return the name corresponding to the node."""
-        return self.out_.op
 
-    @property
-    def inputs(self) -> List[str]:
-        """Return the name of corresponding inputs to the node."""
-        return [x.replace('/', '_').replace('-', '_') for x in self.out_.input]
-
-    @property
-    def node_def_object(self):
-        """Return onnx object."""
-        return self.out_
-
-    @property
-    def tensor_type(self):
-        """Get shape info."""
-        if self.out_.op == 'QTZ_binary_mean_scaling' or \
-                self.out_.op == 'QTZ_linear_mid_tread_half' or \
-                self.out_.op == 'QTZ_binary_channel_wise_mean_scaling':
-            typep = 1
-        else:
-            typep = self.out_.attr["T"].type
-        return typep
-
-    def get_dtype(self):
-        """Get data type info."""
-        dtype_idx = self.tensor_type
-        dtype_str = Input.DATA_TYPE_MAP[dtype_idx]
-        if DLK_DTYPE_MAP[dtype_str] is None:
-            raise UnsupportedDataType(f'Type {dtype_str} is not supported.')
-        return DLK_DTYPE_MAP[dtype_str]  # type: ignore
-
-    def get_shape(self) -> List[str]:
-        """Get shape info."""
-        shape = []
-        for d in self.out_.attr.get('_output_shapes').list.shape:
-            for v in range(len(d.dim)):
-                shape.append(d.dim[v].size)
-
-        return shape[:4]
-
-    def set_shape(self, val: List[str]) -> None:
-        """Set shape info."""
-        raise NotImplemented
+class Output(TFProtoWrapper):
+    def __init__(self, out_nd) -> None:
+        super().__init__(out_nd)
 
 
 class Importer(object):
 
     @classmethod
-    def make_graph(cls, tf_mp) -> Graph:  # type: ignore
+    def make_graph(cls, tf_mp) -> Graph:
```

Comment:
Nice refactoring, but still complicated, not readable.
To encourage refactoring and quality improvement, could you write unit tests of Importer class with several inputs pattern?

Reply:
I think we already have unit test for this in `test_tf_io.py`.

Reply:
I think we should cover all supported operations in test. I prefer to test one or minimum node graph for each supported operations.
But It's ok, Let's do in another PR.